src/Catalyst.Abstractions/Cryptography/ICryptoContext.cs

```diff
@@ -103,6 +103,19 @@ public interface ICryptoContext
         /// <returns></returns>
         ISignature Sign(IPrivateKey privateKey, byte[] message, byte[] context);
 
+        /// <summary>
+        ///     Signs message using provided private key and returns the signature.
+        /// </summary>
+        /// <param name="privateKey"></param>
+        /// <param name="message"></param>
+        /// <param name="messageLength"></param>
+        /// <param name="context"></param>
+        /// <param name="contextLength"></param>
+        /// <returns></returns>
+        ISignature Sign(IPrivateKey privateKey, byte[] message, int messageLength, byte[] context, int contextLength);
+
         bool Verify(ISignature signature, byte[] message, byte[] context);
+        
+        bool Verify(ISignature signature, byte[] message, int messageLength, byte[] context, int contextLength);
```

Comment:
Why not span in such case?

Reply:
I'm trying to Spanify it right now. The good thing is that it will enable to use `ByteString.Span` for free. This means signing/verifying embedded message with no additional allocs.